### PR TITLE
Add GooglePreprocessor and SmartPreprocessor

### PR DESCRIPTION
**Pull request by vemel**
GooglePreprocessor - supports Google format for docstrings
SmartPreprocessor - wisely chooses between RST and Google preprocessors for each docstring

### CODE REVIEW

File: pydocmd/loader.py [hunk 1]
```diff
@@ -31,117 +31,122 @@
...
+            continue
+        new_line = line.lstrip()
+        delta = len(line) - len(new_line)
+        if indent is None:
+            indent = delta
+        elif delta > indent:
+            new_line = ' ' * (delta - indent) + new_line
+        lines[i] = new_line
+
+    return '\n'.join(lines)
```
Comment (NiklasRosenstein) [inline pydocmd/loader.py]: Did the line terminator change here?

Comment (NiklasRosenstein) [reply pydocmd/loader.py]: Ah no the indentation changed. Please go back to two-space, I know it's not common but the rest of the source is currently using a two-space indentation. I'd rather have this PR contain only the changes it intends to address, and not _also_ try to perform cleanup

File: pydocmd/preprocessors/smart.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+from pydocmd.preprocessors.rst import Preprocessor as RSTPreprocessor
+from pydocmd.preprocessors.google import Preprocessor as GooglePreprocessor
+
+
+class Preprocessor(object):
+    """
+    This class implements the preprocessor for restructured text and google.
```
Comment (NiklasRosenstein) [inline pydocmd/preprocessors/smart.py]: I like the idea of a smart preprocessor!

**Comment by NiklasRosenstein** [comment]
Please also be aware that I'm working on a new version that parses the code with `lib2to3` instead of executing the code on the `develop` barnch. It's actually not far from being merged into master.

Anyway, happy to merge your PR into the 2.x code base. I'll likely copy the processor code to the 3.x branch afterwards.

File: pydocmd/loader.py (hunk 1 above)
Comment (vemel) [reply pydocmd/loader.py]: Will do

File: pydocmd/preprocessors/smart.py (hunk 2 above)
Comment (vemel) [reply pydocmd/preprocessors/smart.py]: Works for my project. I did not add simple preprocessor as there is a call of `link_lookup` method that is missing. I could not find it, so I could not test simple preprocessor

Comment (vemel) [reply pydocmd/preprocessors/smart.py]: Reverted pylint auto formatting, now it should be easier to see what has changed

**Comment by vemel** [comment]
Rebased to master, should be no conflicts now.

File: pydocmd/__main__.py [hunk 3]
```diff
@@ -39,244 +39,254 @@
```
Comment (NiklasRosenstein) [inline pydocmd/__main__.py]: Could you please revert these indentation changes as well?

**Comment by NiklasRosenstein** [comment]
Please take a look at the [`pr-84`](https://github.com/NiklasRosenstein/pydoc-markdown/tree/pr-84) branch. I've added some unit tests. Could you take a look at why the smart preprocessor does not detect the Google syntax properly? 😃

File: pydocmd/preprocessors/smart.py (hunk 2 above)
Comment (NiklasRosenstein) [reply pydocmd/preprocessors/smart.py]: Thanks! The link_lookup got introduced in another PR and is a bit ugly at the moment. I believe you would just need to propagate the attribute from the smart processor object to the actual processor that will be used

**Comment by vemel** [comment]
I will take a look, thanks! Actually, I implemented GooglePrerocessor incorrectly, it also supports https://www.python.org/dev/peps/pep-0257 we should probably split these two guys, and we can create a parent class for them as methods will be the same, just param regexps should be different.

FOr now logic for detecting docstring style is quite simple. If I find a section from Google/PEP-0257 - 
 (e.g. `Args:`) I use GooglePreprocessor. Otherwise, RST is used.

File: pydocmd/__main__.py (hunk 3 above)
Comment (vemel) [reply pydocmd/__main__.py]: Done

**Comment by vemel** [comment]
> Please take a look at the [`pr-84`](https://github.com/NiklasRosenstein/pydoc-markdown/tree/pr-84) branch. I've added some unit tests. Could you take a look at why the smart preprocessor does not detect the Google syntax properly?

Fixed, regexp was incorrect.

**Pull request merged**